mupq.py

```diff
@@ -286,20 +341,15 @@ def hash_output(self, output):
     def run_test(self, implementation):
         checksum = self.hash_output(
             super().run_test(implementation).encode('utf-8'))
-        assert self.testvectorhash[implementation.scheme] == checksum
-        self.log.info("Success")
-
-    def run_make(self, target, path, namespace):
-        makeflags = ["make",
-                     f"IMPLEMENTATION_PATH={path}"]
-        if namespace is not None:
-            makeflags.append(f"MUPQ_NAMESPACE={namespace}")
-        makeflags.extend(self.platform_settings.makeflags)
-        makeflags.append(target)
-        return subprocess.check_call(makeflags)
-
+        if self.testvectorhash[implementation.scheme] != checksum:
+            self.log.error("Test %s - %s Failed!", implementation, self.test_type)
+            return -1
```

Comment:
test.py aborts when there is a test failure, while testvectors.py happily continues.
I think it should be consistent. I have a slight preference for the former.

Reply:
I matched the behavior between all tests. I also generally overhauled the error handling a bit. And on the `pqm4` side, I let the entry-scripts exit with a return code if the tests fail.